File: Curso Java Com POO/Java com POO/Aula05/src/teste/ContaBanco.java
package teste;

public class ContaBanco {
    //Atributos
    public int numConta;
    protected String tipo;
    private String dono;
    private float saldo;
    private boolean status;
    
    //Métodos
    public void abrirConta(String tipo) {
        this.setTipo(tipo);
        this.setStatus(true);
        if (tipo.equals ("CC")){
            this.saldo = 50;
        } else if (tipo.equals ("CP")) {
            this.saldo = 150;
        }
    }
    
    public void fecharConta() {
        if (saldo > 0){
            System.out.println("Conta com dinheiro");
        } else if (saldo < 0){
            System.out.println("Conta em débito");
        } else {
            setStatus(false);
        }
    }
    
    public void depositar(float v) {
        if (status = true){
            //saldo = saldo + v;
            setSaldo(getSaldo() + v);
        } else {
            System.out.println("Impossível depositar");
        }      
    }
    
    public void sacar(float v) {
        if (status = true) {
            if (saldo > v) {
                saldo = saldo - v;
            } else {
                System.out.println("Saldo insuficiente");
            }               
        } else {
            System.out.println("Impossível sacar");
        }
    }
    
    public void pagarMensal(){
        int v = 0;
        if (tipo.equals ("CC")) {
            v = 12;
        } else if (tipo.equals ("CP")) {
            v = 20;
        }
        if (status = true) {
            if (saldo > v) {
                saldo = saldo - v;
            } else {
                System.out.println("Saldo insuficiente");
            }                
        } else {
            System.out.println("Impossível sacar");
        }            
    }
    
    //Método Construtor
    public ContaBanco(){
        this.saldo = 0;
        this.status = false;
    }
    
    //Getters e Setters
    //numConta
    public void setNumConta(int numConta){
        this.numConta = numConta;
    }
    public int getNumConta(){
        return this.numConta;
    }
    
    //tipo
    public void setTipo(String tipo){
        this.tipo = tipo;
    }
    public String getTipo(){
        return this.tipo;
    }
    
    //dono
    public void setDono(String dono){
        this.dono = dono;
    }
    public String getDono(){
        return this.dono;
    }
    
    //saldo
    public void setSaldo(float saldo){
        this.saldo = saldo;
    }
    public float getSaldo(){
        return this.saldo;
    }
    
    //status
    public void setStatus(boolean status){
        this.status = status;
    }
    public boolean getStatus(){
        return this.status;
    }
}

File: Curso Programação Orientada a Objetos/PHP com POO/Aula05/ContaBanco.php
<?php

class ContaBanco {
    //Atributos
    public $numConta;
    protected $tipo;
    private $dono;
    private $saldo;
    private $status;
    
    //Métodos
    public function abrirConta($tipo) {
        $this->setTipo($tipo);
        $this->setStatus(true);
        if ($tipo == "CC") {
            $this->setSaldo(50);
        } elseif ($tipo == "CP") {
            $this->setSaldo(150); 
        }
    }
    
    public function fecharConta() {
        if ($this->getSaldo() > 0) {
            echo "<p>Conta ainda tem dinheiro, não posso fechá-la!</p>";
        } elseif ($this->getSaldo() < 0) {
            echo "<p>Conta está em débito. Impossível encerrar!</p>";
        } else {
            $this->setStatus(false);
        }
    }
    
    public function depositar($v) {
        if ($this->getStatus()) {
            $this->setSaldo($this->getSaldo() + $v);
        } else {
            echo "<p>Impossível depositar em um conta fechada!</p>";
        }
    }
    
    public function sacar($v) {
        if ($this->getStatus()) {
            if ($this->getSaldo() >= $v) {
                $this->setSaldo($this->getSaldo() - $v);
            } else {
                echo "<p>Saldo insuficiente para saque!</p>";
            }
        } else {
            echo "<p>Impossível sacar de uma conta fechada!</p>";
        }
    }
    
    public function pagarMensal() {
        if ($this->getTipo() == "CC") {
            $v = 12;
        } elseif ($this->getTipo() == "CP") {
            $v = 20;
        }
        if ($this->getStatus()) {
            if ($this->getSaldo() > $v) {
                $this->setSaldo($this->getSaldo() - $v);
            } else {
                echo "<p>Saldo insuficiente para pagar a mensalidade!</p>";
            }
        } else {
            echo "<p>Impossível pagar mensalidade de uma conta fechada!</p>";
        }
    }
    
    //Construtor
    public function __construct() {
        $this->setSaldo(0);
        $this->setStatus(false);
        echo "<p>Conta criada com sucesso!</p>";
    }
    
    //Setters e Getters
    //numConta
    public function setNumConta($numConta) {
        $this->numConta = $numConta;
    }
    public function getNumConta() {
        return $$this->numConta;
    }
    
    //tipo
    public function setTipo($tipo) {
        $this->tipo = $tipo;
    }
    public function getTipo() {
        return $this->tipo;
    }
    
    //dono
    public function setDono($dono) {
        $this->dono = $dono;
    }
    public function getDono() {
        return $this->dono;
    }
    
    //saldo
    public function setSaldo($saldo) {
        $this->saldo = $saldo;
    }
    public function getSaldo() {
        return $this->saldo;
    }
    
    //status
    public function setStatus($status) {
        $this->status = $status;
    }
    public function getStatus() {
        return $this->status;
    }
}
